### PR TITLE
[release automation] Add documentation building job to build_release_candidate.yml workflow

### PR DESCRIPTION
**Pull request by lostluck**
Create Docs PR with a Github Action for building release artifacts.

Based largely on the existing build_release_candidate.sh script, but with the following changes:

* No longer quarantine each doc generator into it's own clone.
* Leverage available steps for checking out code instead.
* Use `gh` instead of `hub`.
* Adds a use of the release managers' github token.
* Converts parts to steps to clarify logging and failures.
  * Uses local envs, and working directory to clarify where we're working. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @damccorm 

Largely final action, other than the fix to ttypescript (Main in #28133)

**Comment by lostluck** [comment]
I'll update the release guide to go along with this.

File: .github/workflows/build_release_candidate.yml [hunk 1]
```diff
@@ -19,6 +19,9 @@ on:
       APACHE_PASSWORD:
         description: Your Apache password. Required if you want to stage artifacts into https://dist.apache.org/repos/dist/dev/beam/
         required: false
+      BEAM_SITE_TOKEN:
+        description:  Required if you want to create the beam-site docs PR.
```
Comment (damccorm) [inline .github/workflows/build_release_candidate.yml:23]: I assume this is a GitHub Actions PAT? Could we link to https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens and call out the necessary scopes (I assume it needs repo write; anything else?

**Review by damccorm — APPROVED** [review]
Thanks!

File: .github/workflows/build_release_candidate.yml (hunk 1 above)
Comment (lostluck) [reply .github/workflows/build_release_candidate.yml]: That's all my token has. That and "workflow" but I don't think that's necessary here.

If we add the pypi steps to an action we'll get additional use out of the token (same permissions), but that's a later change.

**Comment by lostluck** [comment]
I've removed the additional install line, so typescript will be busted until #28133 is submitted and cherry picked.

**Pull request merged**